play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -1187,6 +1251,229 @@ class RegisterPanel extends PSRoomPanel {
+	static readonly location = 'semimodal-popup';
+	static readonly noURL = true;
+
+	handleOnChange = (ev: Event) => {
+		const setting = "hide" + (ev.currentTarget as HTMLInputElement).name;
+		const value = (ev.currentTarget as HTMLInputElement).checked;
+		let curPref = PS.prefs.chatformatting;
+		curPref[setting] = value;
+		PS.prefs.set("chatformatting", curPref);
+		ev.preventDefault();
+		ev.stopImmediatePropagation();
+	};
+
+	override render() {
+		const room = this.props.room;
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<p>Usable formatting:</p>
+			<p>**<strong>bold</strong>** (<kbd>Ctrl</kbd> + <kbd>B</kbd>)</p>
+			<p>__<em>italics</em>__ (<kbd>Ctrl</kbd> + <kbd>I</kbd>)</p>
+			<p>``<code>code formatting</code>`` (<kbd>Ctrl</kbd> + <kbd>`</kbd>)</p>
+			<p>~~<s>strikethrough</s>~~</p>
+			<p>^^<sup>superscript</sup>^^</p>
+			<p>\\<sub>subscript</sub>\\</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="greentext"
+						checked={PS.prefs.chatformatting.hidegreentext}
+					/> Suppress
+					<span class="greentext">&gt;greentext</span>
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="me"
+						checked={PS.prefs.chatformatting.hideme}
+
+					/> Suppress <code>/me</code>
+					<em>action formatting</em></label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="spoiler"
+						checked={PS.prefs.chatformatting.hidespoiler}
+					/> Auto-show spoilers:
+					<span class="spoiler">these things</span>
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="links"
+						checked={PS.prefs.chatformatting.hidelinks}
+					/> Make [[clickable links]]
+					unclickable
+				</label>
+			</p>
+			<p>
+				<label class="checkbox">
+					<input
+						onChange={this.handleOnChange}
+						type="checkbox"
+						name="interstice"
+						checked={PS.prefs.chatformatting.hideinterstice}
+					/> Don't warn for
+					untrusted links
+				</label>
+			</p>
+			<p><button data-cmd="/close" class="button">Done</button></p>
+		</div>
+		</PSPanelWrapper>;
+	}
+}
+
+class LeaveRoomPanel extends PSRoomPanel {
+	static readonly id = 'confirmleaveroom';
+	static readonly routes = ['confirmleaveroom'];
+	static readonly location = 'semimodal-popup';
+	static readonly noURL = true;
+
+	override render() {
+		const room = this.props.room;
+		const parentRoomId = (this.props.room.parentElem as HTMLInputElement).value;
+		return <PSPanelWrapper room={room} width={480}><div class="pad">
+			<p> Are you sure you want to exit this room?</p>
+			<p class="buttonbar">
+				<button
+					class="button autofocus"
+					name="confirmCloseRoom"
+					value={parentRoomId}
+				>
```

Comment:
```suggestion
				<button data-cmd={`/close ${parentRoomId}`} class="button autofocus">
```
Pretty sure we don't need a separate `confirmCloseRoom` function.